lib/config.ml

```diff
@@ -30,49 +30,42 @@ type t = {
 
 module Opam_repo_fork = struct
   type t = { remote : string; local : Fpath.t }
-end
 
-let of_yaml_exn str =
-  (* ouch *)
-  let lines = String.cuts ~empty:false ~sep:"\n" str in
-  let dict () =
-    List.map
-      (fun line ->
-        match String.cut ~sep:":" line with
-        | Some (k, v) -> (String.trim k, String.trim v)
-        | _ -> failwith "invalid format")
-      lines
-  in
-  let dict = dict () in
-  let find k = try Some (List.assoc k dict) with Not_found -> None in
-  let find_b k =
-    match find k with None -> None | Some s -> Some (bool_of_string s)
-  in
-  let valid = [ "user"; "remote"; "local"; "auto-open"; "keep-v" ] in
-  List.iter
-    (fun (k, _) ->
-      if not (List.mem k valid) then
-        Fmt.failwith "%S is not a valid configuration key." k)
-    dict;
-  let opam_repo_fork =
-    match (find "local", find "remote") with
-    | Some local_str, Some remote ->
-        Fpath.of_string local_str >>| fun local -> (local, remote)
-    | _ ->
-        Rresult.R.error_msg
-          "Dune-release configuration file is compromised: both local and \
-           remote field should be set"
-  in
-  opam_repo_fork >>| fun (local, remote) ->
-  {
-    user = find "user";
-    remote;
-    local;
-    auto_open = find_b "auto-open";
-    keep_v = find_b "keep-v";
-  }
+  let err_msg =
+    Rresult.R.msg
+      "Dune-release configuration file is compromised: both local and remote \
+       field should be set"
+
+  let of_yaml_values dict =
+    let rec aux ((`Remote r, `Local l) as acc) = function
```

Comment:
If I see this right there is not need to unpack the polymorphic variants, since they're only being reconstructed later on.

Reply:
done